exec/java-exec/src/main/java/org/apache/drill/exec/planner/PlannerPhase.java

```diff
@@ -180,9 +180,19 @@ public RuleSet getRules(OptimizerRulesContext context, Collection<StoragePlugin>
 
   PHYSICAL("Physical Planning") {
     public RuleSet getRules(OptimizerRulesContext context, Collection<StoragePlugin> plugins) {
+      final RuleSet storageRules = getStorageRules(context, plugins, this);
+
+      final Builder<RelOptRule> newStorageRulesBuilder = ImmutableSet.builder();
+      for (final RelOptRule relOptRule : storageRules) {
+        // exclude ProjectRemoveRule to prevent Drill from losing column alias or displaying implicit columns
+        if (!(relOptRule instanceof ProjectRemoveRule)) {
```

Comment:
Actually, you are right there is a bug in ProjectRemoveRule in Calcite [1]. Jira was open at the beginning of this year: ProjectRemoveRule loses field names [1].

[1] https://issues.apache.org/jira/browse/CALCITE-1584